secant/Wrappers/WrappedSynchronizer.swift

```diff
@@ -8,26 +8,51 @@
 import Foundation
 import ZcashLightClientKit
 import Combine
+import ComposableArchitecture
+
+enum WrappedSynchronizerState: Equatable {
+    case unknown
+    
+    case transactionsUpdated
+    case started
+    case progressUpdated
+    case statusWillUpdate
+    case synced
+    case stopped
+    case disconnected
+    case syncing
+    case downloading
+    case validating
+    case scanning
+    case enhancing
+    case fetching
+    case minedTransaction
+    case foundTransactions
+    case failed
+    case connectionStateChanged
+}
 
 struct Balance: WalletBalance, Equatable {
     var verified: Int64
     var total: Int64
 }
 
-protocol CombineSynchronizer {
+protocol WrappedSynchronizer {
```

Comment:
I had made this comment but I don't see it anymore. I'm not sure this is the correct name for this protocol. Why is the synchronizer being wrapped? Shouldn't that be reflected in the name of the protocol? 

Wouldn't `ComposableSynchronizer` or something along those lines be more suitable?

Reply:
The naming is challenging, I know. Technically it's "wrapping" the SDKSynchronizer
```swift
class LiveWrappedSynchronizer: WrappedSynchronizer {
    private var cancellables: [AnyCancellable] = []
    private(set) var synchronizer: SDKSynchronizer?
```

`ComposableSynchronizer` is not fit as well, there is no real composable in it. Neither the TCA, neither the composability. This synchronizer is basically a dependency. Object providing interface to get some data in a form of Effects. It is not much different from WrappedMnemonic, WrappedFileManager, etc. It's using some live object iunder the hood to interact with, hence wrapped.

I'll give it today to think something else but WrappedSynchronizer is I believe "the best" for now.

Reply:
After it settled, I think the naming misses the context. What if we add SDK to it? 
```swift
protocol WrappedSDKSynchronizer
```
and then we simply define (exclude wrapped)
```swift
class LiveSDKSynchronizer: WrappedSDKSynchronizer { }

class MockSDKSynchronizer: WrappedSDKSynchronizer { }
```